util/dvsim/dvsim.py

```diff
@@ -36,8 +35,8 @@
 # Try to create the directory if it does not already exist.
 def resolve_scratch_root(arg_scratch_root):
     scratch_root = os.environ.get('SCRATCH_ROOT')
-    if arg_scratch_root == None or arg_scratch_root == "":
-        if scratch_root == None:
+    if not arg_scratch_root:
+        if scratch_root is None:
```

Comment:
Is there a guidance on when to use `is` and when to use `==`?

Reply:
Yes. https://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is

Reply:
For a more canonical answer, the [PEP8](https://www.python.org/dev/peps/pep-0008) text is pretty readable. In particular, see the "Programming Recommendations" section.